components/motor/tmcstepper/stepper_motor_tmc.go

```diff
@@ -148,18 +140,29 @@ const (
 )
 
 // NewMotor returns a TMC5072 driven motor.
-func NewMotor(ctx context.Context, deps resource.Dependencies, c TMC5072Config, name resource.Name,
-	logger logging.Logger,
+func newMotor(ctx context.Context, deps resource.Dependencies, c resource.Config, logger logging.Logger,
 ) (motor.Motor, error) {
-	bus := buses.NewSpiBus(c.SPIBus)
-	return makeMotor(ctx, deps, c, name, logger, bus)
+	conf, err := resource.NativeConfig[*TMC5072Config](c)
+	if err != nil {
+		return nil, err
+	}
+	bus := buses.NewSpiBus(conf.SPIBus)
+	return makeMotor(ctx, deps, *conf, c.ResourceName(), logger, bus)
 }
 
 // makeMotor returns a TMC5072 driven motor. It is separate from NewMotor, above, so you can inject
 // a mock SPI bus in here during testing.
 func makeMotor(ctx context.Context, deps resource.Dependencies, c TMC5072Config, name resource.Name,
 	logger logging.Logger, bus buses.SPI,
 ) (motor.Motor, error) {
+	if c.MaxRPM == 0 {
+		logger.CWarn(ctx, "max_rpm not set, setting to 300 rpm")
+		c.MaxRPM = 200
```

Comment:
either change the warning to 200 or change c.MaxRPM to 300

Reply:
ah thanks for catching forgot to change both